src/cli/dot_command.zig

```diff
@@ -0,0 +1,169 @@
+            return if (run) tryToRun(ctx, file, "index.js") else RunOutput{ .cancelled = false, .script = "index.js" };
+        } else if (returnFileIfExist("index.ts")) |file| {
+            return if (run) tryToRun(ctx, file, "index.ts") else RunOutput{ .cancelled = false, .script = "index.ts" };
+        } else if (returnFileIfExist("index.mjs")) |file| {
+            return if (run) tryToRun(ctx, file, "index.mjs") else RunOutput{ .cancelled = false, .script = "index.mjs" };
+        } else if (returnFileIfExist("index.cjs")) |file| {
+            return if (run) tryToRun(ctx, file, "index.cjs") else RunOutput{ .cancelled = false, .script = "index.cjs" };
+        } else if (returnFileIfExist("index.mts")) |file| {
+            return if (run) tryToRun(ctx, file, "index.mts") else RunOutput{ .cancelled = false, .script = "index.mts" };
+        } else if (returnFileIfExist("index.cts")) |file| {
+            return if (run) tryToRun(ctx, file, "index.cts") else RunOutput{ .cancelled = false, .script = "index.cts" };
+        }
+
+        return if (run) RunOutput{ .cancelled = true, .script = "index.js" } else RunOutput{ .cancelled = false, .script = null };
+    }
+
+    // From cli.zig - maybeOpenWithBunJS but modified to accept file & script_to_run
+    fn tryToRun(ctx: *Command.Context, file: std.fs.File, script_to_run: string) RunOutput {
+        ctx.args.entry_points = &[_]string{script_to_run};
+
+        var script_name_buf: [bun.MAX_PATH_BYTES]u8 = undefined;
+
+        Global.configureAllocator(.{ .long_running = true });
+
+        var absolute_script_path = std.os.getFdPath(file.handle, &script_name_buf) catch return RunOutput{ .cancelled = true, .script = script_to_run };
+        Run.boot(
+            ctx.*,
+            file,
+            absolute_script_path,
+        ) catch |err| {
+            if (Output.enable_ansi_colors) {
+                ctx.log.printForLogLevelWithEnableAnsiColors(Output.errorWriter(), true) catch {};
+            } else {
+                ctx.log.printForLogLevelWithEnableAnsiColors(Output.errorWriter(), false) catch {};
+            }
+
+            Output.prettyErrorln("<r><red>error<r>: Failed to run <b>{s}<r> due to error <b>{s}<r>", .{
+                std.fs.path.basename(ctx.args.entry_points[0]),
+                @errorName(err),
+            });
+
+            Global.exit(1);
+        };
+
+        return RunOutput{ .cancelled = false, .script = script_to_run };
+    }
+
+    pub fn exec(ctx: *Command.Context) !bool {
+        var ORIGINAL_PATH: string = "";
+        var this_bundler: bundler.Bundler = undefined;
+        var root_dir_info = try RunCommand.configureEnvForRun(ctx.*, &this_bundler, null, &ORIGINAL_PATH, true);
+
+        var script_to_run: ?string = "";
+        if (root_dir_info.enclosing_package_json) |package_json| script_to_run = package_json.main_fields.get("module") orelse package_json.main_fields.get("main") orelse null;
```

Comment:
The error message should include the chosen main field name

Reply:
![image](https://user-images.githubusercontent.com/56601352/188316683-9edb1fc2-c2c0-4894-9d2a-3eeb0c0b8721.png)
Module name is main field